drivers/ccs811/ccs811_saul.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ccs811
+ * @brief       SAUL adaption for AMS CCS811 digital gas sensor devices
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "phydat.h"
+#include "saul.h"
+
+#include "ccs811.h"
+
+static uint16_t _iaq_tvoc = 0;
+static uint16_t _iaq_eco2 = 0;
+
+static int read(const ccs811_t *dev)
+{
+    /* check whether new data can be read */
+    int res = ccs811_data_ready((ccs811_t *)dev);
+
+    if (res != CCS811_OK) {
+        return res;
+    }
+
+    /* read new data and save them to local storage */
+    return ccs811_read_iaq((ccs811_t *)dev, &_iaq_tvoc, &_iaq_eco2, 0, 0);
+}
+
+static int read_tvoc(const void *dev, phydat_t *res)
+{
+    /* read new data if available */
+    read(dev);
```

Comment:
Here nothing is done with the return value of `read`. So it fails, there's no way to know about it.

Reply:
The approach was different. Local variables _iaq_tvoc and _iaq_eco2 were returned. Even though function read fails, values are given back.

I have changed it.